src/client/pages/AssessmentSection/DataTable/hooks/useGetTableData.ts

```diff
@@ -20,30 +20,67 @@ const dependencyTables: Record<string, Array<string>> = {
   [TableNames.extentOfForest]: [TableNames.valueAggregate],
 }
 
-const useTableNames = (props: Props): Array<string> => {
+type Dependency = { tableName: TableName; cycleName?: CycleName; assessmentName?: AssessmentName }
+type Depencies = Array<Dependency>
+
+const useDependencies = (props: Props): Depencies => {
   const { sectionName, table } = props
   const canEdit = useCanEdit(sectionName)
 
   const { calculationDependencies, validationDependencies } = table
   const { name: tableName } = table.props
 
-  const tableNames: Array<string> = useMemo<Array<string>>(() => {
-    const tableNamesSet = new Set<string>([tableName])
-    dependencyTables[tableName]?.forEach((t) => tableNamesSet.add(t))
+  const tableNames: Depencies = useMemo<Depencies>(() => {
+    const dependencies: Depencies = [
+      {
+        tableName,
+      },
+    ]
+
+    dependencyTables[tableName]?.forEach((t) =>
+      dependencies.push({
+        tableName: t,
+      })
+    )
 
     if (canEdit && validationDependencies) {
       Object.values(validationDependencies).forEach((variables) =>
-        variables.forEach((variable) => tableNamesSet.add(variable.tableName))
+        variables.forEach((variable) => {
+          const exists = dependencies.some(
+            (dependency) =>
+              dependency.tableName === variable.tableName &&
+              dependency.cycleName === variable.cycleName &&
+              dependency.assessmentName === variable.assessmentName
+          )
+          if (!exists)
+            dependencies.push({
+              tableName: variable.tableName,
+              cycleName: variable.cycleName,
+              assessmentName: variable.assessmentName,
+            })
+        })
       )
     }
 
     if (calculationDependencies) {
       Object.values(calculationDependencies).forEach((variables) =>
-        variables.forEach((variable) => tableNamesSet.add(variable.tableName))
+        variables.forEach((variable) => {
+          const exists = dependencies.some(
+            (dependency) =>
+              dependency.tableName === variable.tableName &&
+              dependency.cycleName === variable.cycleName &&
+              dependency.assessmentName === variable.assessmentName
+          )
+          if (!exists)
+            dependencies.push({
+              tableName: variable.tableName,
+              cycleName: variable.cycleName,
+              assessmentName: variable.assessmentName,
+            })
+        })
       )
```

Comment:
The two iterations above look the same . Any chance to improve the maintenability of the code above ?